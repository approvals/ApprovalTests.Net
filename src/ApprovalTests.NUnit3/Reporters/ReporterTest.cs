using ApprovalTests.Reporters.ContinuousIntegration;

[TestFixture]
public class ReporterTest
{
    [Test]
    public void Testname()
    {
        var old = Environment.GetEnvironmentVariable(NCrunchReporter.EnvironmentVariable);
        Environment.SetEnvironmentVariable(NCrunchReporter.EnvironmentVariable, "1");
        Assert.IsTrue(NCrunchReporter.INSTANCE.IsWorkingInThisEnvironment("a.txt"));
        Environment.SetEnvironmentVariable(NCrunchReporter.EnvironmentVariable, old);
    }

    [Test]
    public void TestInvalidReporterShouldThrow()
    {
        var attribute = new UseReporterAttribute(typeof(ReporterTest));
        VerifyReporterAttribute(attribute);
    }

    [Test]
    public void TestMultipleWithInvalidReporterShouldThrow()
    {
        var attribute = new UseReporterAttribute(typeof(ReporterTest), typeof(string));
        VerifyReporterAttribute(attribute);
    }

    static void VerifyReporterAttribute(UseReporterAttribute attribute)
    {
        var reporter = (IEnvironmentAwareReporter) attribute.Reporter;
        var reportException = Assert.Throws<Exception>(() => reporter.Report("a.txt", "a.txt"));
        var isWorkingException = Assert.Throws<Exception>(() => reporter.IsWorkingInThisEnvironment("a.txt"));

        Approvals.Verify($"""
                          {reportException.Message}

                          {isWorkingException.Message}
                          """);
    }
}