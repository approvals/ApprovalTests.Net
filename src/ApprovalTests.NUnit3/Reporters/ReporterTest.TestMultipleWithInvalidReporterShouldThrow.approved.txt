Invalid configuration of reporter. Reporters must extend IApprovalFailureReporter.
Invalid reporter type: System.String

Note: The stack here is not helpful.

Invalid configuration of reporter. Reporters must extend IApprovalFailureReporter.
Invalid reporter type: ReporterTest

Note: The stack here is not helpful.